syntax = "proto3";

// option java_multiple_files = true;
// option java_package = "io.grpc.examples.routeguide";
// option java_outer_classname = "RouteGuideProto";
// option objc_class_prefix = "RTG";

package chatservice;

// // Interface exported by the server.
// service ChatService {
//   rpc MyMethod(MyRequest) returns (MyResponse) {}
// }

// message MyRequest {
//   string message = 1;
// }

// message MyResponse {
//   string reply = 1;
// }

service AuthService {
  rpc Login (LoginRequest) returns (LoginResponse) {}
  rpc Register (RegisterRequest) returns (RegisterResponse) {}
}

service ChatService {
  rpc Send (SendRequest) returns (SendResponse) {}
  // rpc Receive
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  bool success = 1;
  string message = 2;
}

message RegisterRequest {
  string username = 1;
  string password = 2;
}

message RegisterResponse {
  bool success = 1;
  string message = 2;
}

message SendRequest {
  string sender = 1;
  string receiver = 2;
  string body = 3;
}

message SendResponse {
  bool success = 1;
  string message = 2;
}
